backend/src/controllers/auth/register.controller.ts

```diff
@@ -0,0 +1,15 @@
+import { Request, Response, NextFunction } from 'express'
+import { createUser } from './register.services'
+
+import { BadRequestError } from '../../errors'
+
+export const registerUser = async (req: Request, res: Response) => {
+    const { username, email, password } = req.body
+    if (!username || !email || !password) {
+        throw new BadRequestError('Please provide all values')
+    }
+    
+    const user = await createUser({ username, email, password })
+    
+    res.status(201).json({ user })
+}
```

Comment:
Update this to `User.controller.ts`
```suggestion
import { Request, Response, NextFunction } from 'express'
import * as userService from './register.services'

import { IRegisterRq } from './auth.interface'

export const registerUser = async (req: Request, res: Response) => {
    const rq: IRegisterRq = req.body
    
    userService.validate(rq);

    const user = await userService.createUser(rq);
    res.status(201).json({ user })

}
```